### PR TITLE
Refactors lib/private/Accounts classes

### PR DESCRIPTION
**Pull request by fsamapoor**
Following [previous PRs taking advantage of PHP8's constructor property promotion](https://github.com/nextcloud/server/pulls?q=is%3Apr+author%3Afsamapoor+Uses+PHP8%27s+constructor+property+promotion) in `/core/` namespace, I have also made the required adjustments to the classes in `/lib/private/Accounts` namespace.

The improvements in this PRs include but are not limited to:
- Using PHP8's constructor property promotion
- Adding return types
- Adding types to properties
- Removing redundant docblocks

### CODE REVIEW

**Review by nfebe — APPROVED** [review]
Looks okay! +1 on the more precise match.

Thank you

**Comment by solracsf** [comment]
CI is failing.

**Comment by fsamapoor** [comment]
@solracsf Thank you for the heads up. I wasn't aware that the failing CI was in fact a result of failing test cases. I have fixed the failing tests and everything seems to be fine now.

File: lib/private/Accounts/AccountManager.php [hunk 1]
```diff
@@ -138,35 +107,22 @@ class AccountManager implements IAccountManager {
...
-		ICrypto                  $crypto
+		private IDBConnection $connection,
+		private IConfig $config,
+		private EventDispatcherInterface $eventDispatcher,
+		private IJobList $jobList,
+		private LoggerInterface $logger,
+		private IVerificationToken $verificationToken,
+		private IMailer $mailer,
+		private Defaults $defaults,
+		private IFactory $factory,
```
Comment (blizzz) [inline lib/private/Accounts/AccountManager.php:118]:
```suggestion
		private IFactory $l10nFactory,
```

rename this for readbility/clarity, factory alone is a bit broad

**Review by blizzz — CHANGES_REQUESTED** [review]
small nitpick

**Review by blizzz — APPROVED** [review]

File: lib/private/Accounts/AccountManager.php (hunk 1 above)
Comment (fsamapoor) [reply lib/private/Accounts/AccountManager.php]: @blizzz Thanks. Other than making it more readable, I searched the codebase and almost all other classes also used this convention. So keeping it consistent is always good.

**Comment by blizzz** [comment]
failing tests unrelated => merging

**Pull request merged**